influxql/ast.go

```diff
@@ -1296,14 +1296,25 @@ func (s *SelectStatement) String() string {
 }
 
 // RequiredPrivileges returns the privilege required to execute the SelectStatement.
-func (s *SelectStatement) RequiredPrivileges() ExecutionPrivileges {
-	ep := ExecutionPrivileges{{Admin: false, Name: "", Privilege: ReadPrivilege}}
+func (s *SelectStatement) RequiredPrivileges() (ExecutionPrivileges, error) {
+	ep := ExecutionPrivileges{}
+	for _, source := range s.Sources {
+		measurement, ok := source.(*Measurement)
+		if !ok {
+			return nil, fmt.Errorf("invalid measurment: %s", source)
+		}
+
+		ep = append(ep, ExecutionPrivilege{
+			Name:      measurement.Database,
```

Comment:
Just making a note here, this seems like an error because measurements that haven't had their databases specified will not have been filled in yet, but it ends up not being an error because the query authorizer fills in a blank database field with the currently queried database.

A comment here regarding that little caveat might be helpful for someone else. I almost got tricked and thought this didn't work.